cloud/amazon/ec2_lc.py

```diff
@@ -212,6 +221,15 @@ def create_launch_config(connection, module):
         classic_link_vpc_id=classic_link_vpc_id,
     )
 
+    if append_hash:
+        # MD5 of launch configuration properties
+        h = hashlib.md5()
+        h.update(str(frozenset(lc.__dict__)))
+
+        # Update name variables with md5 hash
+        name = '-'.join((name, h.hexdigest()))
```

Comment:
This is mostly an aesthetic request, but the AWS UI doesn't handle super-long launchconfig names that well. Could you maybe truncate the hash to 8 characters? For the hash-collision-concerned: 8 hex characters encodes 32 bits of the hash, and to reach a probability of 50% collision, you'd need to have (on average) 77164 hashes. Of course, it's unlikely someone will have 77,000 launchconfigs, let alone 77,000 at the same time. 

```
hash_size = 2**32 # 32 bits
probUnique = 1.0
for k in xrange(1, 200000):
    probUnique = probUnique * (hash_size - (k - 1)) / hash_size
    if probUnique <= 0.5: 
        print "greater than 50% probability of collision at {} hashes, probability {}".format(k, probUnique * 100)
        break
```
```